### PR TITLE
Automatically switch difftype if other has files and the current one doesn't

### LINKED ISSUE #32: Auto-detect only staged changes on refresh
If there are no changes in the working tree and there are staged changes, automatically switch to staged changes on refresh

### PR DESCRIPTION
**Pull request by NotNotQuinn**
closes #32

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -57,7 +57,27 @@ impl MyApp {
...
+                            if app_data.staged_diff_data.stats.files_changed == 0
+                                && app_data.modified_diff_data.stats.files_changed != 0
+                            {
+                                self.control_data.diff_type = data::DiffType::Modified
+                            }
+                        }
+                    };
+
+                    self.app_data = Some(app_data);
+                }
```
Comment (M4tthewDE) [inline src/main.rs:80]: This should happen in UI code. I want to keep logic out of the message handling as much as possible.

**Review by M4tthewDE — CHANGES_REQUESTED** [review]

**Comment by NotNotQuinn** [comment]
I don't really like this solution. It "works", but you can't open changes that don't have files in them. This doesn't logistically create any problems but it removes a feeling of control when using the application. Read previous commit message for more information.

I have a few ideas for resolutions:
 - Place the diff type update logic in `Message::LoadDiff` or another `Message::*` (old)
 - Update the diff type every frame (current)
 - Add a control flag (`control_data.just_loaded_diff`) which is set in `Message::LoadDiff`
 - New method `MyApp::on_load_diff()` called in `Message::LoadDiff`

**Comment by M4tthewDE** [comment]
As I see it, we only care about the initial selection, before the user has selected anything. The ui should display whatever is inside of the ControlData. I would add another function to MyApp, which is called when Message::LoadDiff is received.
Inside the function the AppData is created and the correct initial DiffType is computed.

File: src/main.rs [hunk 2]
```diff
@@ -57,7 +78,10 @@ impl MyApp {
                         Err(_) => s.send(Message::ShowError("Error loading diff!".to_string())),
                     });
                 }
-                Message::UpdateAppData(app_data) => self.app_data = Some(app_data),
+                Message::UpdateAppData(app_data) => {
+                    self.app_data = Some(app_data);
+                    self.update_app_data();
+                }
```
Comment (M4tthewDE) [inline src/main.rs:84]: Pass the app_data from the Message directly to the function. That way you don't have to check inside of self.update_app_data() if it's not None.

**Review by M4tthewDE — APPROVED** [review]
LGTM

**Pull request merged**